packages/ses/src/error/assert.js

```diff
@@ -53,7 +53,28 @@ freeze(quote);
  */
 const hiddenDetailsMap = new WeakMap();
 
-/** @type {DetailsTag} */
+// TODO Move this type declaration to types.js as a separate @callback type,
+// without breaking the meaning of the type. As currently written, if it is
+// moved into a separate @callback type, it no longer understands that `args`
+// is a rest parameter. I have not yet figured out how to declare that it is,
+// except by having it here directly annotating the `details` function.
```

Comment:
I have mixed feelings about `@callback`.

I'd recommend moving it to `types.js` and declaring it as something like:
```js
/** @typedef {(template: TemplateStringsArray, ...args: Array<any>) => ...} DetailsTag */
```

Reply:
Oh, this is diff churn. @erights added the `DetailsTag` type to `types.js` already and this is a holdover from before I rebased.

I’m removing this text. Though, for posterity, I think that `@callback` is useful in JSDoc over what appears in `types.js` because it produces an opportunity to describe each parameter with prose to go with the type and name. I look forward to using jsdoc for docs too!